src/components/Categories.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import './../css/categories.css';
+
+// product type def - need to make global
+interface Product {
+    id: string;
+    image: string;
+    title: string;
+    price: number;
+    category: string;
+    description: string;
+  }
+
+function Categories() {
+    const [categories, setCategories] = useState([]);
+    const [selectedCategory, setSelectedCategory] = useState('all');
+
+    const [products, setProducts] = useState<Product[]>([]);
+
+    // fetch array of categories for selectable filters
+    useEffect(() => {
+        async function fetchCategories() {
+        try {
+            const response = await fetch('https://fakestoreapi.com/products/categories');
+            const data = await response.json();
+            setCategories(data);
+        } catch (error) {
+            console.error('Error fetching categories:', error);
+        }
+    }
+
+        fetchCategories();
+    }, []);
+
+    // fetch array of product objects for display based on current filter
+    useEffect(() => {
+        async function fetchProducts() {
+        try {
+            const response = await fetch('https://fakestoreapi.com/products');
+            const data: Product[] = await response.json();
+            setProducts(data);
+        } catch (error) {
+            console.error('Error fetching products:', error);
+        }
+        }
+
+        fetchProducts();
+    }, []);
```

Comment:
I know this is a temp thing to be eventually replaced by React-Query. But if you wanted to get real fancy you could have a custom hook for fetches
```Typescript
const data = useFetch<Product[]>('https://fakestoreapi.com/products')
```
Within it you could have this useEffect and a `data` useState that is returned which is the `response.json()`

Or alternatively have an onSuccess callback
```Typescript
useFetch<Product[]>({
  url: 'https://fakestoreapi.com/products',
  onSuccess: (data) => setProducts(data),
})
```